src/main/java/edu/kis/powp/jobs2d/command/CommandEditor.java

```diff
@@ -0,0 +1,47 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class CommandEditor {
+
+    private DriverCommand command;
+
+    CommandEditor(DriverCommand command) {
+        this.command = command;
+    }
+
+    public void changeCoordinates(int x, int y) throws ImproperCommandTypeException {
+        if (command instanceof ICompoundCommand) {
+            throw new ImproperCommandTypeException();
+        }
+        if (command instanceof SetPositionCommand) {
+            command = new SetPositionCommand(x, y);
+        } else if (command instanceof OperateToCommand) {
+            command = new OperateToCommand(x, y);
+        }
```

Comment:
instanceof is a code smell of a bad OO design. Try to eliminate it. Maybe visitor can be used?